### ISSUE TITLE
Problems initializing StepLR

### ISSUE THREAD

**Issue by carlosfranzreb** [NVIDIA/NeMo#4774]
**Describe the bug**

I am using a pre-trained Conformer Transducer and want to initialize an optimizer and an LR scheduler using its method `setup_optimization`, which returns the optimizer and the LR scheduler.

The problem I am having is that the ASR model's method `setup_optimization` method requires `max_steps` to be part of the scheduler config. Otherwise, it doesn't initialize the scheduler. But given that  `StepLR` does not accept `max_steps` as an argument, it raises an error.

**Steps/Code to reproduce bug**

```python
from nemo.collections.asr.models.rnnt_bpe_models import EncDecRNNTBPEModel

config = {
    "name": "adamw",
    "lr": 0.1,
    "betas": [0.9, 0.98],
    "weight_decay": 0.001,
    "sched": {"name": "StepLR", "step_size": 1, "last_epoch": 1},
}

model = EncDecRNNTBPEModel.from_pretrained("stt_de_conformer_transducer_large")
model.setup_optimization(config)
```

Running this script outputs the following log:

```
[NeMo W 2022-08-19 10:30:46 lr_scheduler:816] Neither `max_steps` nor `iters_per_batch` were provided to `optim.sched`, cannot compute effective `max_steps` !
    Scheduler will not be instantiated !
```

When adding `max_steps` with an integer value (e.g. `"max_steps". 1`) to `config["sched"]`, the following exception occurs:

```
Exception has occurred: TypeError
__init__() got an unexpected keyword argument 'max_steps'
  File "/home/carlos/repos/accsr/lr_issue.py", line 12, in <module>
    model.setup_optimization(config)
```

**Expected behavior**

StepLR does not require a `max_steps` argument and should be initialized without it.

**Environment overview (please complete the following information)**

 - Environment location: Local machine, conda environment
 - Method of NeMo install: pip install (version 1.10.0)
 
**Environment details**

If NVIDIA docker image is used you don't need to specify these.
Otherwise, please provide:
- OS version: Ubuntu 20.04
- PyTorch version: 1.12.0
- Python version: 3.8.0

**Comment by titu1994** [comment]
See the execution flow here - https://github.com/NVIDIA/NeMo/tree/main/examples/asr/asr_transducer#model-execution-overview

1) set trainer
2) setup train dataset
3) then call setup optimization

**Comment by titu1994** [comment]
Since it doesn't require it, you'll need to setup the train data + trainer and it will skip calculation of max steps

**Comment by carlosfranzreb** [comment]
Thanks for replying!

Unfortunately, my use case differs from the one shown in the diagram. I am looking for a function that returns the optimizer and the scheduler given the config shown above, both for NeMo schedulers and also those from PyTorch. `setup_optimization` works fine if the scheduler takes the `max_steps` argument, but it doesn't otherwise.

**Comment by titu1994** [comment]
That's because setup optimization is not supposed to work in isolation. It depends on the output of previous steps - ie how many global steps exist and the setup of the trainer.

**Comment by carlosfranzreb** [comment]
Alright, then I will look for a workaround. Thank you!
